bin/uglifyjs

```diff
@@ -227,24 +227,20 @@ function run() {
             result.ast.figure_out_scope({});
         }
         print(JSON.stringify(result.ast, function(key, value) {
-            switch (key) {
+            if (value) switch (key) {
               case "thedef":
-                if (typeof value == "object" && typeof value.id == "number") {
-                    return value.id;
-                }
-                return;
+                return symdef(value);
               case "enclosed":
                 return value.map(function(sym){
-                    return sym.id;
+                    return symdef(sym);
                 });
               case "variables":
               case "functions":
               case "globals":
-                if (value && value.size()) {
-                    var ret = {};
+                if (value.size()) {
+                    var ret = [];
                     value.each(function(val, key) {
-                        // key/val inverted for readability.
-                        ret[val.id] = key;
+                        ret.push(symdef(val));
```

Comment:
Sorry for nitpicking - would you mind using `Dictionary.map`?
https://github.com/mishoo/UglifyJS2/blob/edb4e3bd52e1623425927a7d63963ba3b87a3ec2/lib/utils.js#L300-L305

Reply:
done